src/west/app/project.py

```diff
@@ -556,14 +556,22 @@ def do_run(self, args, ignored):
         self._setup_logging(args)
 
         failed = []
+        # We may need to force git to use colors if the user wants them,
+        # which it won't do ordinarily since stdout is not a terminal.
+        color = ['--color=always'] if log.use_color() else []
+
         for project in self._cloned_projects(args):
-            log.banner(f'diff for {project.name_and_path}:')
             # Use paths that are relative to the base directory to make it
             # easier to see where the changes are
-            try:
-                project.git(['diff', f'--src-prefix={project.path}/',
-                             f'--dst-prefix={project.path}/'])
-            except subprocess.CalledProcessError:
+            cp = project.git(['diff', f'--src-prefix={project.path}/',
+                              f'--dst-prefix={project.path}/',
+                              '--exit-code'] + color,
+                             capture_stdout=True, capture_stderr=True,
+                             check=False)
+            if cp.returncode == 1 or log.VERBOSE > log.VERBOSE_NONE:
+                log.banner(f'diff for {project.name_and_path}:')
+                log.inf(cp.stdout.decode('utf-8'))
```

Comment:
are we sure users want this ?

If I run `west diff` today, I might see:
```
=== diff for zephyr (zephyr):
=== diff for nffs (modules/fs/nffs):
=== diff for segger (modules/debug/segger):
=== diff for mbedtls (modules/crypto/mbedtls):
....
```
but if we remove those lines per default, users might be unsure if the command was actually executed in all repos.
At minimum inform something like:
```
=== no diff for zephyr (zephyr):
=== no diff for nffs (modules/fs/nffs):
=== no diff for segger (modules/debug/segger):
=== no diff for mbedtls (modules/crypto/mbedtls):
....
```

But maybe even better could be to collect repos with no diff, and then print a final message, such as:
```
=== No diff in projects: zephyr, nffs, segger, mbedtls, ....
```
I don't think the path is important for projects without changes.

Reply:
Good idea. Updated. It now prints something like this:

```
$ west  diff
=== diff for manifest (zephyr):
diff --git zephyr/scripts/dts/gen_defines.py zephyr/scripts/dts/gen_defines.py
index 4270eb7b4f..2d3d33a309 100755
--- zephyr/scripts/dts/gen_defines.py
+++ zephyr/scripts/dts/gen_defines.py
@@ -431,7 +431,7 @@ def phandle_macros(prop, macro):
             out_comment(f"No label macro emitted for {prop.node.name} "
                         f"property {prop.name}:\n"
                         f"{prop.val.path} has no label\n",
-                        blank_before=False)
+                        blank_before=False)  # hello
     else:
         for i, node in enumerate(prop.val):
             ret[f"{macro}_IDX_{i}_VAL_LABEL"] = quote_str(node.label)

Empty diff in 31 projects.
```